apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/spring/webmvc/SpringTransactionNameInstrumentation.java

```diff
@@ -134,8 +134,10 @@ public static void setName(Transaction transaction, String className, @Nullable
                 name.append(className);
                 if (methodName != null) {
                     name.append('#').append(methodName);
+                } else { 
+                	name.append("#").append(transaction.withName(className, PRIO_LOW_LEVEL_FRAMEWORK + 1));
```

Comment:
That doesn't look quite right yet. You're adding the `className` twice. Could you add a test case?

If the method name is present, set the transaction name should set to `${className}#${methodName}` with the priority `PRIO_HIGH_LEVEL_FRAMEWORK`. Otherwise, set the transaction name to just the `className` with the priority `PRIO_LOW_LEVEL_FRAMEWORK + 1`

Reply:
Hey @felixbarny i have addressed your comment and made changes accordingly. Can you please review again to check the logic. Henceforth, i will add test cases.